### PR TITLE
UCT/MAKEFILE: Enhance the check for the libmlx5 version on the host.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by alinask** [comment]
@yosefe please review

File: src/uct/Makefile.am [hunk 1]
```diff
@@ -14,7 +14,7 @@ libuct_la_CPPFLAGS = $(BASE_CPPFLAGS) -I$(abs_top_srcdir)/src -I$(abs_top_buildd
 libuct_la_LDFLAGS  = -ldl -version-info $(SOVERSION)
 libuct_la_LIBADD   = $(LIBM) ../ucs/libucs.la
 libuct_ladir       = $(includedir)/uct
-libmlx5_ver        = \"`rpm -qf $(IBVERBS_DIR)/include/infiniband/mlx5_hw.h &>/dev/null && rpm -qf /usr/include/infiniband/mlx5_hw.h`\"
+libmlx5_ver        = \"`(rpm -qf $(IBVERBS_DIR)/include/infiniband/mlx5_hw.h 2>/dev/null || rpm -qf /usr/include/infiniband/mlx5_hw.h 2>/dev/null) | head -1`\"
```
Comment (miked-mellanox) [inline src/uct/Makefile.am]: it will break debian/ubuntu distros which have no rpm but other pkg manager

Comment (yosefe) [reply src/uct/Makefile.am]: how will it break? errors are redirected to devnull

Comment (miked-mellanox) [reply src/uct/Makefile.am]: maybe logic should be as following:
* extract mofed/libmlx5 version with grep from /bin/ofed_info:
```
$head -10 /bin/ofed_info
#!/bin/bash
if [ "$1" == "-s" ]; then echo MLNX_OFED_LINUX-3.3-1.0.0.0:; exit; fi
cat << EOF
MLNX_OFED_LINUX-3.3-1.0.0.0 (OFED-3.3-1.0.0):

...
```

Comment (miked-mellanox) [reply src/uct/Makefile.am]:
```
$head  -2 /bin/ofed_info|grep MLNX|awk '{print $9}'| sed -e 's/:;//g'
MLNX_OFED_LINUX-3.3-1.0.0.0
```

Comment (miked-mellanox) [reply src/uct/Makefile.am]: on system w/o rpm, rpm not found, empty command redirected to "|head -1"

Comment (yosefe) [reply src/uct/Makefile.am]: @miked-mellanox not sure we want to assume OFED existence
worst case, on system w/o rpm, libmlx5_ver will be empty, so the user error message will not be very useful

Comment (miked-mellanox) [reply src/uct/Makefile.am]: mlx5_hw.h exists only in mofed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5956/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
Imho, looks like a very horrible practice to me.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5958/ for details (Mellanox internal link).

**Pull request merged**